### PR TITLE
Fix stCelo compatibility

### PR DESCRIPTION
**Pull request by martinvol**
Fixes https://github.com/celo-org/celo-blockchain-planning/issues/642

### CODE REVIEW

File: packages/protocol/contracts-0.8/common/EpochManager.sol [hunk 1]
```diff
@@ -494,4 +513,12 @@ contract EpochManager is
       CELOequivalent
     );
   }
+
+  function _setElectedSigners() internal {
+    IAccounts accounts = getAccounts();
+    electedSigners = new address[](elected.length);
+    for (uint i = 0; i < elected.length; i++) {
```
Comment (pahor167) [inline packages/protocol/contracts-0.8/common/EpochManager.sol]: Im quite worried about adding another loop into this function

**Comment by soloseng** [comment]
For stCelo this approach works, but it does not really bring backwards compatibility to all other contracts in the CELO ecosystem. 

Any contracts that are making use of `UsingPrecompiles` that were previously deployed by members of the CELO community will break. This also potentially includes Mento

File: packages/protocol/contracts-0.8/governance/Validators.sol [hunk 2]
```diff
@@ -1054,8 +1054,8 @@ contract Validators is
    * @param value New reset period for slashing multiplier.
    */
   function setSlashingMultiplierResetPeriod(uint256 value) public nonReentrant onlyOwner {
-    allowOnlyL1();
     slashingMultiplierResetPeriod = value;
+    emit SlashingMultiplierResetPeriodSet(value);
```
Comment (soloseng) [inline packages/protocol/contracts-0.8/governance/Validators.sol:1058]: Slashing multiplier is no longer used by any L2 contracts. is there a reason for keeping them active?

File: packages/protocol/contracts-0.8/governance/Validators.sol [hunk 3]
```diff
@@ -611,7 +613,6 @@ contract Validators is
    * @param validatorAccount The validator to deaffiliate from their affiliated validator group.
    */
   function forceDeaffiliateIfValidator(address validatorAccount) external nonReentrant onlySlasher {
-    allowOnlyL1();
```
Comment (martinvol) [inline packages/protocol/contracts-0.8/governance/Validators.sol:614]: TODO probably needs to stay removed https://github.com/celo-org/celo-monorepo/pull/11225

**Comment by martinvol** [comment]
Changes have been cherry picked by https://github.com/celo-org/celo-monorepo/pull/11236

**Pull request closed without merging**